tenseal/tensealcontext.cpp

```diff
@@ -243,6 +248,11 @@ bool TenSEALContext::auto_mod_switch() {
     return this->_auto_flags & flag_auto_mod_switch;
 }
 
+bool TenSEALContext::equals(const std::shared_ptr<TenSEALContext>& other) {
+    return this->parms() == other->parms() &&
+           this->is_public() == other->is_public();
+}
```

Comment:
Maybe we can give this a deeper thought, what about the other attributes, the scale for example? And I don't understand why the contexts needs to be both the same type (public vs private)

Reply:
It's enough to compare the public key and scale? I am not sure about the relin/galois/secret keys here.